src/zarr/core/array.py

```diff
@@ -461,8 +461,11 @@ async def create(
             raise ValueError("Only one of chunk_shape or chunks can be provided.")
 
         if chunks:
+            chunks = parse_shapelike(chunks)
             _chunks = normalize_chunks(chunks, shape, dtype_parsed.itemsize)
         else:
+            if chunk_shape:
+                chunk_shape = parse_shapelike(chunk_shape)
```

Comment:
I suggest we move this check into `normalize_chunks`. Here's where the behavior you're looking to change is:

https://github.com/zarr-developers/zarr-python/blob/2fe12a7f424ac78c18a688d0a75049a3df67c98f/src/zarr/core/chunk_grids.py#L116-L117

If you look at `normalize_chunks`, you'll see why Zarr is more permissive on chunk shape than array shape. I'm not opposed to us disallowing floats but we need to be careful keep some of the other behaviors in `normalize_chunks`.

Reply:
That seems like a better place indeed! I've updated the PR.